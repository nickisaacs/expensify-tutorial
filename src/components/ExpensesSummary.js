import React from 'react';
import {connect} from 'react-redux';
import getTotalExpenses from '../selectors/expenses-total';
import getVisibleExpenses from '../selectors/expenses';
import numeral from 'numeral';

export const ExpensesSummary = ({total,count}) =>(
    <div>
        <h1>Viewing {count} {count === 1 ? 'expense' : 'expenses'} totalling  {numeral(total/100).format('$0,0.0')}</h1>
    </div>
);

const mapStateToProps = (state) => {
    const visibleExpenses = getVisibleExpenses(state.expenses,state.filters)
    return {
        total : getTotalExpenses(visibleExpenses),
        count: visibleExpenses.length
    };
};

export default connect(mapStateToProps)(ExpensesSummary);


