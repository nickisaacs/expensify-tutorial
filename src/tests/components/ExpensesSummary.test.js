import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';

test('Should correctly render Expenses Summary with 1 expense',() => {
    const wrapper = shallow(<ExpensesSummary count={1} total={235}/>);
    expect(wrapper).toMatchSnapshot();

});

test('Should correctly render Expenses Summary with multiple expenses',() => {
    const wrapper = shallow(<ExpensesSummary count={6} total={23432325}/>);
    expect(wrapper).toMatchSnapshot();
});