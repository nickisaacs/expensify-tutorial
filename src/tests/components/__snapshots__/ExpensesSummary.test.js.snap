// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render Expenses Summary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling  
    $2.4
  </h1>
</div>
`;

exports[`Should correctly render Expenses Summary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    6
     
    expenses
     totalling  
    $234,323.3
  </h1>
</div>
`;
